### PR TITLE
Check laws via purescript-quickcheck-laws

### PR DESCRIPTION
**Pull request by hdgarrood**
I used freebroccolo's fork for now, because it builds with 0.7.0.

### CODE REVIEW

File: check-laws/bower.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "purescript-maybe-check-laws",
+  "private": "true",
+  "dependencies": {
+    "purescript-maybe": "./..",
```
Comment (garyb) [inline check-laws/bower.json]: I had no idea you could even do that. Nice!

**Comment by garyb** [comment]
Well this makes a lot of sense. The relative dependency basically fixes all the issues I had with maintaining two `bower.json`s. I'll hang fire on merging it for now though, as I'd like to fix up `purescript-quickcheck-laws` "properly" first. I should be able to spend some time on that tonight.

**Comment by hdgarrood** [comment]
Rebased and updated to use the published version of `purescript-quickcheck-laws`. Unfortunately the distributivity law doesn't always fail, and even when it does, it's usually only on 1 out of the 1000 tests. This might be a symptom of a poor generator, or an indication that we need to increase the number of tests.

**Comment by garyb** [comment]
I found that the tests exited early when attempting to raise the number of iterations past 1000 unfortunately, as yeah, I'd much rather it failed sooner too. I'm not too sure why it doesn't, but I don't think randomness is the problem, I seem to recall when I used strongcheck it failed similiarly infrequently.

Either way we should remove the `MonadPlus` instance as it is indeed broken. We can re-add when the `MonadZero` based stuff is sorted out.

**Comment by garyb** [comment]
We should definitely come back to doing this somehow, maybe with `psc-package` when it's released - I just discovered #24 using `quickcheck-laws`.

**Pull request closed without merging**